src/client/src/hooks/useDeleteEntityPrompts.tsx

```diff
@@ -0,0 +1,158 @@
+import { useContext } from "react";
+import { useTranslation } from "react-i18next";
+import { useHistory } from "react-router-dom";
+import { Trash2, X } from "lucide-react";
+import { User, Workgroup } from "../api/apiInterfaces.ts";
+import { deleteUser } from "../api/user.ts";
+import { deleteWorkgroup } from "../api/workgroup.ts";
+import { PromptContext } from "../components/prompt/promptContext.tsx";
+import { useApiRequest } from "./useApiRequest.ts";
+
+export const useDeleteEntityPrompts = <T extends User | Workgroup>(
+  deleteEntity: (id: number) => Promise<void>,
+  setSelectedEntity: (entity: T | null) => void,
+  entities: T[],
+  setEntities: (entities: T[]) => void,
+) => {
+  const history = useHistory();
+  const { t } = useTranslation();
+  const { showPrompt } = useContext(PromptContext);
+  const { callApiWithRollback } = useApiRequest();
+
+  // Type guard for User
+  function isUser(entity: User | Workgroup): entity is User {
+    return "subjectId" in entity;
+  }
+
+  // Type guard for Workgroup
+  function isWorkgroup(entity: User | Workgroup): entity is Workgroup {
+    return "boreholeCount" in entity;
+  }
+
+  const getNotDeletableMessage = (entity: User | Workgroup) => {
+    let notDeletableMessage = "";
+    if (isWorkgroup(entity)) {
+      notDeletableMessage += `${t("msgDisablingWorkgroup")}.`;
+      if (!entity?.isDisabled) {
+        notDeletableMessage += ` ${t("msgReenablingWorkgroupTip")}.`;
+      }
+    }
+    if (isUser(entity)) {
+      notDeletableMessage += `${t("msgDisablingUser")}.`;
+      if (!entity?.isDisabled) {
+        notDeletableMessage += ` ${t("msgReenablingUserTip")}.`;
+      }
+    }
+    return notDeletableMessage;
+  };
+
+  const getDeletableMessage = (entity: User | Workgroup) => {
+    let deletableMessage = "";
+    if (isWorkgroup(entity)) {
+      deletableMessage += `${t("deleteWorkgroupMessage")}`;
+      if (!entity?.isDisabled) {
+        deletableMessage += ` ${t("msgReenablingWorkgroupTip")}.`;
+      }
+    }
+    if (isUser(entity)) {
+      deletableMessage += `${t("deleteUserMessage")}`;
+      if (!entity?.isDisabled) {
+        deletableMessage += ` ${t("msgReenablingUserTip")}.`;
```

Comment:
Q/PP: Dieser gesamte Code ist für zwei Objekte bereits ziemlich komplex bzw. ähnlich (→ Code-Duplikate). Wenn in Zukunft weitere Objekttypen hinzukommen, wird das bisschen hässlich. Können die Translations dynamisch aufgebaut werden? So dass es z.B. `msgDisabling` gibt und man die Entität (z.B. `t("workgroup")` oder `t("user")`) dann mitgibt? So wie bei https://github.com/swisstopo/swissgeol-boreholes-suite/blob/d0f1be5d4b6302e4c1cad31be1675cc37e3b27d2/src/client/public/locale/de/common.json#L192.

Reply:
Gute Idee!